tests/run-make/reproducible-build/rmake.rs

```diff
@@ -0,0 +1,222 @@
+        rustc().arg("-g").input("reproducible-build-aux.rs").run();
+        rustc()
+            .arg("-g")
+            .input("reproducible-build.rs")
+            .linker(&cwd().join(bin_name("linker")).display().to_string())
+            .run();
+        rustc()
+            .arg("-g")
+            .input("reproducible-build.rs")
+            .linker(&cwd().join(bin_name("linker")).display().to_string())
+            .run();
+        diff().actual_file("linker-arguments1").expected_file("linker-arguments2").run();
+    });
+
+    run_in_tmpdir(|| {
+        rustc().input("linker.rs").opt().run();
+        rustc().opt().input("reproducible-build-aux.rs").run();
+        rustc()
+            .opt()
+            .input("reproducible-build.rs")
+            .linker(&cwd().join(bin_name("linker")).display().to_string())
+            .run();
+        rustc()
+            .opt()
+            .input("reproducible-build.rs")
+            .linker(&cwd().join(bin_name("linker")).display().to_string())
+            .run();
+        diff().actual_file("linker-arguments1").expected_file("linker-arguments2").run();
+    });
+
+    run_in_tmpdir(|| {
+        rustc().input("reproducible-build-aux.rs").run();
+        rustc().input("reproducible-build.rs").crate_type("rlib").library_search_path("b").run();
+        rfs::copy(rust_lib_name("reproducible_build"), rust_lib_name("foo"));
+        rustc().input("reproducible-build.rs").crate_type("rlib").library_search_path("a").run();
+        assert_eq!(rfs::read(rust_lib_name("reproducible_build")), rfs::read(rust_lib_name("foo")));
+    });
+
+    run_in_tmpdir(|| {
+        rustc().input("reproducible-build-aux.rs").run();
+        rustc()
+            .input("reproducible-build.rs")
+            .crate_type("rlib")
+            .arg("--remap-path-prefix=/a=/c")
+            .run();
+        rfs::copy(rust_lib_name("reproducible_build"), rust_lib_name("foo"));
+        rustc()
+            .input("reproducible-build.rs")
+            .crate_type("rlib")
+            .arg("--remap-path-prefix=/b=/c")
+            .run();
+        assert_eq!(rfs::read(rust_lib_name("reproducible_build")), rfs::read(rust_lib_name("foo")));
+    });
+
+    run_in_tmpdir(|| {
+        rustc().input("reproducible-build-aux.rs").run();
+        rfs::create_dir("test");
+        rfs::copy("reproducible-build.rs", "test/reproducible-build.rs");
+        rustc()
+            .input("reproducible-build.rs")
+            .crate_type("bin")
+            .arg(&format!("--remap-path-prefix={}=/b", cwd().display()))
+            .run();
+        eprintln!("{:#?}", rfs::shallow_find_dir_entries(cwd()));
+        rfs::copy(bin_name("reproducible_build"), bin_name("foo"));
+        rustc()
+            .input("test/reproducible-build.rs")
+            .crate_type("bin")
+            .arg("--remap-path-prefix=/test=/b")
+            .run();
+        assert_eq!(rfs::read(bin_name("reproducible_build")), rfs::read(bin_name("foo")));
```

Comment:
Neat, thank you very much, but that seems like a flaw in the `run_make_support` library. Do you think `bin_name` should be modified to apply this normalization?

Reply:
I don't think so, because doing or not doing hyphen normalization is tool-specific. For rustc, bin output artifact name derived from the input file name with no output file name explicitly specifued happens to not perform such normalization.